students/JoeNunnelley/session03/list_lab.py

```diff
@@ -0,0 +1,157 @@
+#! /usr/bin/env python3
+
+"""
+List Lab Excercises
+Author : Joe Nunnelley
+"""
+def series1():
+    """
+    When the script is run, it should accomplish the following four series of actions:
+    Series 1
+
+        Create a list that contains “Apples”, “Pears”, “Oranges” and “Peaches”.
+        Display the list (plain old print() is fine…).
+        Ask the user for another fruit and add it to the end of the list.
+        Display the list.
+        Ask the user for a number and display the number back to the user and the
+        fruit corresponding
+        to that number (on a 1-is-first basis). Remember that Python uses zero-based
+        indexing, so you will need to correct.
+        Add another fruit to the beginning of the list using “+” and display the list.
+        Add another fruit to the beginning of the list using insert() and display the list.
+        Display all the fruits that begin with “P”, using a for loop.
+    """
+    print()
+    print('>> Series 1 <<')
+    print()
+    fruit = ['Apples', 'Pears', 'Orange', 'Peaches']
+    print(fruit)
+    response = input('Add fruit > ')
+    fruit.append(response)
+    print(fruit)
+    response = input('Get fruit number >')
+    try:
+        print('{} => {}'.format(response, fruit[int(response)-1]))
+    except IndexError as identifier:
+        print("Index not found in list: {}\n{}".format(int(response) - 1, identifier))
+
+    fruit = ["Lychee"] + fruit
+    print(fruit)
+    fruit.insert(0, "Guava")
+    print(fruit)
+
+    for _ in fruit:
+        if "P" in str.upper(_):
+            print(_)
+
+    return fruit
+
+
+def series2(fruits_in):
+    """
+    Series 2
+
+    Using the list created in series 1 above:
+
+        Display the list.
+        Remove the last fruit from the list.
+        Display the list.
+        Ask the user for a fruit to delete, find it and delete it.
+        (Bonus: Multiply the list times two. Keep asking until a match is found.
+        Once found, delete all occurrences.)
+    """
+    print()
+    print('>> Series 2 <<')
+    print()
+    print(fruits_in)
+    fruits_in.pop()
+    print(fruits_in)
+    response = input('Which fruit to delete >')
+
+    try:
+        fruits_in.remove(response)
+    except ValueError as identifier:
+        print("Did not find this item: {}\n{}".format(response, identifier))
```

Comment:
you generally don't want to print the exception like that -- it is exposing the code, which the user doesn't care about -- but I like the message otherwise.